src/Stack.cpp

```diff
@@ -0,0 +1,78 @@
+#include "Stack.h"
+
+Stack::Stack(const size_t& initial_size, const int& initial_value) : m_data(Vector(initial_size, initial_value)) {}
+
+void Stack::push(int value) {
+    m_data.push_back(value);
+}
+
+void Stack::pop() {
+    m_data.pop_back();
+}
+
+int &Stack::top() {
+    return m_data[m_data.size() - 1];
+}
+
+const int &Stack::top() const {
+    return m_data[m_data.size() - 1];
+}
+
+bool Stack::empty() const {
+    return m_data.size() == 0;
+}
+
+size_t Stack::size() const {
+    return m_data.size();
+}
+
+bool Stack::operator==(const Stack &other) const {
+    return m_data.size() == other.m_data.size();
+}
+
+bool Stack::operator!=(const Stack &other) const {
+    return m_data.size() != other.m_data.size();
+}
+
+Stack Stack::operator+(int value) const {
+    Stack result{};
+    result.m_data.reserve(m_data.size() + 1);
+    result.m_data = m_data;
```

Comment:
This is a very bad idea. What if `result` is deleted? This will cause `this`'s pointer to be freed and can cause horrible bugs. You definitely should move pointers around like that.
If you want to create a new `Stack` object you must copy its contents